### ISSUE TITLE
Ask for password if user adds a point not located near their current location

### ISSUE THREAD

**Issue by craigappl** [OpenSRP/opensrp-client-reveal#42]
A user has the ability to add a point in the Android client. Issue https://github.com/OpenSRP/opensrp-client-reveal/issues/40 adds the ability to define a buffer server side. We would like to require a user to enter a password if they are not physically close to the structure they add a point.

- [x] Check current location of user when they touch the button to add a point
- [x]   Implement logic: If a user is within the validation buffer, we let them proceed as normal. If the user is not within the validation buffer, ask for their password.
- [ ] This option needs to be enabled server side and synced down to the Android client (one global configuration for all users) using the same configuration that's defined in https://github.com/OpenSRP/opensrp-client-reveal/issues/41.

**Comment by githengi** [comment]
@craigappl currently on mspray the users would enter a global password for the campaign and not their own password. This is a check to ensure that is done by super user kind of users

Kindly confirm if we should adopt this

**Comment by craigappl** [comment]
Yes, we should adopt this in OpenSRP. There are a few things to note:
1. We will set a single global password for the MVP
2. The size of the radius should be configurable in the server and expand at different zoom levels
3. We should have the ability to turn off this feature as a configuration variable
4. The password should be able to be changed without requiring a server restart
5. We don't need a UI for the password for the MVP, just the ability for a system administrator to change it.

We should do our best to make sure that these configurations don't require a server restart or release of the APK, but are defined server side and synced down.

**Comment by githengi** [comment]
> 3\. We should have the ability to turn off this feature as a configuration variable

When this is disabled then the behavior is be to disable interacting with structures that are further than the buffer or allow the interaction without needing any password?

**Comment by craigappl** [comment]
@githengi When the feature is turned off, we should allow users to interact with any structure on the map.

**Comment by githengi** [comment]
@craigappl When this feature is enabled, and location services are disabled or we are unable to determine users location, then we should ask for password to be able to add points on open spray forms?
